src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java

```diff
@@ -202,6 +186,89 @@ public void stopContainer(String containerName) {
         dockerClient.execStartCmd(createStopContainerFileCmdResponse.getId()).exec(new ResultCallback.Adapter<>());
     }
 
+    /**
+     * Copy the repositories and build script from the Artemis container to the build job container.
+     *
+     * @param buildJobContainerId        the id of the build job container
+     * @param assignmentRepositoryPath   the path to the assignment repository
+     * @param testRepositoryPath         the path to the test repository
+     * @param auxiliaryRepositoriesPaths the paths to the auxiliary repositories
+     * @param auxiliaryRepositoriesNames the names of the auxiliary repositories
+     * @param buildScriptPath            the path to the build script
+     */
+    public void populateBuildJobContainer(String buildJobContainerId, Path assignmentRepositoryPath, Path testRepositoryPath, Path[] auxiliaryRepositoriesPaths,
+            String[] auxiliaryRepositoriesNames, Path buildScriptPath) {
+        copyToContainer(assignmentRepositoryPath.toString(), buildJobContainerId);
+        renameDirectoryOrFile(buildJobContainerId, assignmentRepositoryPath.getFileName().toString(), "assignment-repository");
```

Comment:
I think the code would be a bit cleaner if this were extracted to a method

```suggestion
        copyToBuildJobContainer(buildJobContainerId, assignmentRepositoryPath, "assignment-repository")
```